### PR TITLE
STORM-1040. SQL support for Storm

### PR DESCRIPTION
**Pull request by haohui**
This is a duplicate of #914.

This PR rebases all the changes in the STORM-1040 branch towards the latest master.

### CODE REVIEW

**Comment by harshach** [comment]
still +1 from me.

**Comment by darionyaphet** [comment]
+1

File: external/sql/README.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+# Storm SQL
+
+Compile SQL queries to Storm topologies.
+
+## Usage
+
+Run the ``storm sql`` command to compile SQL statements into Trident topology, and submit it to the Storm cluster
+
+```
+$ bin/storm sql <topo-name> <sql-file>
```
Comment (arunmahadevan) [inline external/sql/README.md]: minor: can you have `topo-name` as the last argument to be in sync with the `storm jar` command where topology name is the last argument?

**Comment by unsleepy22** [comment]
LGTM +1, good work!

**Comment by Parth-Brahmbhatt** [comment]
+1.

**Comment by wuchong** [comment]
nice work! +1

**Comment by harshach** [comment]
Thanks @haohui . Merged into master.

**Pull request merged**